src/CodeGen.fs

```diff
@@ -923,113 +927,39 @@ type StringEnumAttribute() =
-        | Some false -> "\n        <CopyLocalLockFileAssemblies>false</CopyLocalLockFileAssemblies>"
-
-    sprintf """<Project Sdk="Microsoft.NET.Sdk">
-    <PropertyGroup>
-        <TargetFramework>netstandard2.0</TargetFramework>
-        <LangVersion>latest</LangVersion>%s
-    </PropertyGroup>
-    <ItemGroup>
-%s
-    </ItemGroup>
-    <ItemGroup>
-        <PackageReference Update="FSharp.Core" Version="4.7.2"/>
-        <PackageReference Include="Fable.Remoting.Json" Version="2.14.0" />
-    </ItemGroup>
-</Project>
-    """ attribute files
-
-let sampleSharedProject files =
-    sprintf """<Project Sdk="Microsoft.NET.Sdk">
-    <PropertyGroup>
-        <TargetFramework>netstandard2.0</TargetFramework>
-        <LangVersion>latest</LangVersion>
-    </PropertyGroup>
-    <ItemGroup>
-%s
-    </ItemGroup>
-    <ItemGroup>
-        <PackageReference Update="FSharp.Core" Version="4.7.2" />
-    </ItemGroup>
-</Project>
-"""  files
-
-let sampleSharedFSharpProject project copyLocalLockFileAssemblies =
-    let attribute =
-        match copyLocalLockFileAssemblies with
-        | None -> String.empty
-        | Some true -> "\n        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>"
-        | Some false -> "\n        <CopyLocalLockFileAssemblies>false</CopyLocalLockFileAssemblies>"
-
-    sprintf """<Project Sdk="Microsoft.NET.Sdk">
-    <PropertyGroup>
-        <TargetFramework>netstandard2.0</TargetFramework>
-        <LangVersion>latest</LangVersion>%s
-    </PropertyGroup>
-    <ItemGroup>
-        <Compile Include="%s.GraphqlClient.fs" />
-    </ItemGroup>
-    <ItemGroup>
-        <PackageReference Update="FSharp.Core" Version="4.7.2" />
-        <PackageReference Include="Fable.Remoting.Json" Version="2.14.0" />
-        <ProjectReference Include="..\shared\%s.Shared.fsproj" />
-    </ItemGroup>
-</Project>
-"""  attribute project project
-
-let sampleSharedFableProject project =
-    sprintf """<Project Sdk="Microsoft.NET.Sdk">
-    <PropertyGroup>
-        <TargetFramework>netstandard2.0</TargetFramework>
-        <LangVersion>latest</LangVersion>
-    </PropertyGroup>
-    <ItemGroup>
-        <Compile Include="%s.GraphqlClient.fs" />
-    </ItemGroup>
-    <ItemGroup>
-        <Content Include="*.fsproj; *.fs; *.js" Exclude="**\*.fs.js" PackagePath="fable\" />
-    </ItemGroup>
-    <ItemGroup>
-        <PackageReference Update="FSharp.Core" Version="4.7.2" />
-        <PackageReference Include="Fable.SimpleHttp" Version="3.0.0" />
-        <PackageReference Include="Fable.SimpleJson" Version="3.19.0" />
-        <ProjectReference Include="..\shared\%s.Shared.fsproj" />
-    </ItemGroup>
-</Project>
-"""  project project
+let generateProjectDocument
+    (packageReferences: XElement seq)
+    (files: XElement seq)
+    (copyLocalLockFileAssemblies: bool option)
+    (contentItems: XElement seq)
+    (projectReferences: XElement seq)=
```

Comment:
This is entirely optional but I think it could make the code more readable if this function takes a record as input so that the call-site looks like this
```fs
CodeGen.generateProjectDocument {
  nugetPackageReferences = references
  files = files
  copyLocalLockFileAssemblies = config.CopyLocalLockFileAssemblies
  contentItems = content
  projectReferences = projects
}
```
because the parameters are mostly `XElement seq`